bears/general/AnnotationBear.py

```diff
@@ -28,104 +27,212 @@ def run(self, filename, file, language: str, coalang_dir: str = None):
                                 respectively.
         """
         lang_dict = LanguageDefinition(language, coalang_dir=coalang_dir)
-        # Strings
-        # TODO treat single-line and multiline strings differently
-        strings = dict(lang_dict["string_delimiters"])
-        strings.update(lang_dict["multiline_string_delimiters"])
-        strings_found = self.find_with_start_end(filename, file, strings)
-        # multiline Comments
-        comments_found = self.find_with_start_end(
-                filename, file, dict(lang_dict["multiline_comment_delimiters"]))
-        # single-line Comments
-        comments_found.update(self.find_singleline_comments(
-                          filename, file, list(lang_dict["comment_delimiter"])))
-
-        matches_found = strings_found | comments_found
-        # Remove Nested
-        unnested_annotations = set(filter(
-                                       lambda arg: not starts_within_ranges(
-                                           arg, matches_found),
-                                       matches_found))
-        # Yield different results for strings and comments
-        strings_found = tuple(filter(lambda arg: arg in unnested_annotations,
-                                     strings_found))
-        comments_found = tuple(filter(lambda arg: arg in unnested_annotations,
-                                      comments_found))
-        yield HiddenResult(self, {'comments': comments_found,
-                                  'strings': strings_found})
+        string_delimiters = dict(lang_dict["string_delimiters"])
+        multiline_string_delimiters = dict(
+            lang_dict["multiline_string_delimiters"])
+        multiline_comment_delimiters = dict(
+            lang_dict["multiline_comment_delimiters"])
+        comment_delimiter = dict(lang_dict["comment_delimiter"])
+        strings = comments = ()
```

Comment:
these could also be called something like string_ranges and comment_ranges tbh. Don't be afraid of longer names if they are more descriptive